### DISCUSSION TITLE
Cypress and VPNs

### DISCUSSION THREAD

**Discussion by StarGateMiner** [cypress-io/cypress#15987] [Questions and Help]
Hi team, is it possible to use Cypress with VPNs, 

So my issues is, I conduct for example regression test on registration forms, some forms are blocked in specific countries e.g. Spain. I then manually need to put on VPN switch to Spain for the test to run.

Is there a way to eliminate this issue and get Cypress to automatically switch on/off VPN software or a plugin that maybe able to help? Any ideas, specially userful when running automation tests for country blocks.

Thanks.

**Comment by portyu9** [comment]
Hi @StarGateMiner 

Cypress does not control VPN software directly. The VPN/network location should be configured **outside Cypress**, before starting the test run.

For geo-specific tests, the cleanest approach is to run Cypress through a Spain-based CI runner/proxy/VPN connection, for example:

```bash
vpn-connect spain
npx cypress run
vpn-disconnect
```

If your VPN provides a CLI, you can automate those commands in your CI script or invoke an external process with `cy.task()`, but switching network routes during an active browser session can be unreliable.

If a geo-proxy is sufficient, Cypress also supports `HTTP_PROXY` / `HTTPS_PROXY`:

```bash
HTTPS_PROXY=http://spain-proxy:port npx cypress run
```

So the recommended fix is: **select the required country at the OS/CI/proxy layer, then launch Cypress for that country-specific test run.**

**Reply by portyu9** [reply]
@StarGateMiner Please mark this answer as accepted if it helped, thank you.
